src/compiler/parser.ts

```diff
@@ -1589,9 +1605,10 @@ namespace Parser {
             return result;
         }
 
-        initializeState(fileName, sourceText, languageVersion, syntaxCursor, scriptKind);
+        skipJSDoc = !!skipJSDoc && scriptKind !== ScriptKind.JS && scriptKind !== ScriptKind.JSX;

```

Comment:
Until reading this line, I was really confused as to why the compiler host has a property `skipJSDoc` that gets unconditionally passed into the parser, when the compiler host might be asked to produce a JS file that definitely needs its JSDoc parsed. I think I would prefer a longer, more explicit name like `skipJSDocInTsFiles` or something.

Reply:
I was considering instead writing `canSkipJSDoc` to say "it's safe to skip if you'd like"; is that better?

skipJSDocInTsFiles isn't quite accurate either, since it still has to parse comments which contain `@see` or `@link` as those affect import elision and usage checks (boo).

Reply:
Also totally happy to bikeshed this in general; I totally agree that `skipJSDoc` is not descriptive enough. I also thought of `needsJSDoc` but that's the wrong direction; defaulting to false is a lot easier to deal with IMO.

Reply:
`canSkipJSDoc` is better, though I personally think `skipJSDocInTsFiles` is close enough even with the caveat about `@see` and `@link`.

Reply:
Bikesheding more; the only caveat with `skipJSDocInTsFiles` is that I can't stick a `Parsing` suffix on it to make the `Host` property clearer. I had intended to make the name descriptive enough that it was clear what it did / didn't do, but inside the scanner/parser I just said `skipJSDoc` because when you're in the parser, every option has to do with parsing so `skipJSDocParsing` was redundant.

But maybe the longer name is already clear enough that my concern is moot.

Reply:
Per group chat, `skipNonSemanticJSDoc` and `skipNonSemanticJSDocParsing` are what seemed to win for some definition of "win". Kinda good because I already called them "semantic" when I named the regex.